### PR TITLE
Remove astropy-healpix

### PR DESCRIPTION
**Pull request by cdeil**
This PR removes the unused `astropy-healpix` dependency for `regions`.

@bmatthieu3 @astrofrog - OK?

---

This dependency was added in #212 because the plan was to (partly) merge MOCPy into regions.
But after discussing with @astrofrog and @bmatthieu3 at PyGammapy19, we decided to keep MOCPy separate for now (see https://github.com/astropy/regions/pull/219#issuecomment-492264190), which means there is no need for HEALPix in `regions`.

To summarise, we now have for the near future this:
- `regions` (independent, no dependency on `astropy-healpix`)
- `astropy-healpix` (independent, no dependency on `regions`)
- `mocpy` (depends on astropy-healpix, and maybe will depend on `regions` if @bmatthieu3 exposes the MOC there as a SkyRegion subclass)

### CODE REVIEW

**Comment by bmatthieu3** [comment]
@cdeil - That is a good idea to inherit spatial MOC from SkyRegion in `mocpy`.

@astrofrog, @cdeil - I am currently moving the healpix dependency of mocpy from `astropy_healpix` to `cdshealpix`. I changed the python-rust wrapper of `cdshealpix` following our pygamma discussion to pyo3 and I am now using the rust-numpy crate that allows me to directly pass numpy view arrays from python to rust code! The new version of cdshealpix (v0.2.0) comprising these changes has been deployed. What I still need to do for the future is comparing the results between cdshealpix and astropy_healpix.

**Comment by cdeil** [comment]
> I am currently moving the healpix dependency of mocpy from astropy_healpix to cdshealpix.

What was our conclusion concerning how to proceed with `astropy_healpix` and `cdshealpix` development? Let's continue the discussion in the `astropy-healpix` issue tracker, can you find a good issue for that, or open a new one, please?

> I changed the python-rust wrapper of cdshealpix following our pygamma discussion to pyo3 and I am now using the rust-numpy crate that allows me to directly pass numpy view arrays from python to rust code! The new version of cdshealpix (v0.2.0) comprising these changes has been deployed.

Nice! I started learning Rust a few days ago. I'll check it out for sure in the coming weeks.

> What I still need to do for the future is comparing the results between cdshealpix and astropy_healpix.

OK. But don't spend too much time on this. Especially concerning precision we know what `astropy-healpix` isn't good, there are known issues: https://github.com/astropy/astropy-healpix/issues?q=is%3Aissue+is%3Aopen+label%3Aprecision
And known solutions proposed by FX and Martin Reinecke to get good precision.
So overall, spending time comparing precision against the current `astropy-healpix` is not time well-spent IMO, we should just move ahead to a new implementation. Adding more unit tests for `astropy-healpix` before doing that, with reference values e.g. from the C++ lib or somewhere else would be useful of course, they can be added and marked with an `xfail(reason="TODO: needs precise algo")`. That would make it easier to change / improve backend at any future time.

**Comment by bmatthieu3** [comment]
> Let's continue the discussion in the astropy-healpix issue tracker, can you find a good issue for that, or open a new one, please?

Yes. I opened a new issue on astropy-healpix: astropy/astropy-healpix#128
